src/scipp/core/binning.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from typing import Dict, List, Optional, Union
+
+from .._scipp import core as _cpp
+from .variable import array, Variable, linspace, arange
+from .bins import bin, histogram
+
+
+def _get_coord(x, name):
+    event_coord = x.bins.meta.get(name) if x.bins is not None else None
+    return x.meta.get(name, event_coord)
+
+
+def _upper_bound(x):
+    import numpy as np
+    bound = x.max()
+    bound.value = np.nextafter(bound.value, np.inf)
+    return bound
+
+
+def _parse_coords_arg(x, name, arg):
+    coord = _get_coord(x, name)
```

Comment:
`coord` can be `None`, at least add a TODO for better error handling.

Reply:
Added an exception.